mobile-app/www/js/app.js

```diff
@@ -87,60 +120,74 @@ var app = {
-      if (code) code = code[1];
+        authWindow.addEventListener('loadstart', function(e) {
+          var code = new RegExp(app.redirectUri + '\\?code=([^&]+)?').exec(e.url);
+          if (code) code = code[1];
 
-      var error = new RegExp(app.redirectUri + '\\?error=([^&]+)').exec(e.url);
-      if (error) error = error[1];
+          var error = new RegExp(app.redirectUri + '\\?error=([^&]+)').exec(e.url);
+          if (error) error = error[1];
 
-      if (code || error) {
-        authWindow.close();
-      }
+          if (code || error) {
+            authWindow.close();
+          }
+        });
+
+      } else {
+        // Web page, restore browser's original window.open because cordova-inappbrowser-plugin overwrote this
+
+        // check we already have a code
+        var code = new RegExp(app.redirectUri + '\\?code=([^&]+)?').exec(document.URL);
+        if (code) code = code[1];
+        var error = new RegExp(app.redirectUri + '\\?error=([^&]+)').exec(document.URL);
+        if (error) error = error[1];
+        if (!(code || error)) {
+          myWindowOpen(authUrl, '_self', 'location=no,toolbar=no');
+        }
+    }
+
+    if (error) {
+      app.showNotification("Error: " + error);
+    }
 
-      if (code) {
-        $$.ajax({
-          url: 'https://slack.com/api/oauth.access' +
-            '?client_id=' + app.clientId +
-            '&client_secret=' + app.clientSecret +
-            '&code=' + code,
-          dataType: 'json',
-          success: function (data) {
+    if (code) {
+      $$.ajax({
+        url: 'https://slack.com/api/oauth.access' +
+          '?client_id=' + app.clientId +
+          '&client_secret=' + app.clientSecret +
+          '&code=' + code,
+        dataType: 'json',
+        success: function (data) {
+          if (data.access_token != undefined) { 
             app.token = data.access_token;
-            app.saveToken(fileEntry, app.token);
+            app.saveToken(app.token);
             app.showNotification("Slack authentication successful");
+          } else {
+            if (data.ok == false) {
+              app.showNotification("Slack get auth token error: " + data.error);
+            }
           }
-        });
-      }
-
-      if (error) {
-        app.showNotification("Error: " + error);
-      }
-    });
+        }
+      });
+    }
   },
 
-  saveToken: function(fileEntry, token) {
-    fileEntry.createWriter(function (fileWriter) {
-      fileWriter.write(token);
-    });
+  saveToken: function(token) {
+      window.localStorage.setItem(app.slackAuthTokenKeyName,token);
   },
 
   onDeviceReady: function() {
-    window.requestFileSystem(LocalFileSystem.PERSISTENT, 1024, function (fs) {
-      fs.root.getFile("token", { create: true, exclusive: false }, function (fileEntry) {
-        app.getToken(fileEntry);
-      });
-    });
-
+    app.initiateSlackAuth();
     app.showNotification("Device ready");
```

Comment:
I would also recommend removing this notification, for the same reasons as above.